pinot-core/src/main/java/com/linkedin/pinot/core/util/DataTableCustomSerDe.java

```diff
@@ -0,0 +1,444 @@
+import com.linkedin.pinot.core.query.aggregation.function.MinMaxRangeAggregationFunction;
+import com.linkedin.pinot.core.query.aggregation.function.quantile.digest.QuantileDigest;
+import com.linkedin.pinot.core.segment.creator.impl.V1Constants;
+import it.unimi.dsi.fastutil.doubles.DoubleArrayList;
+import it.unimi.dsi.fastutil.ints.IntOpenHashSet;
+import java.io.ByteArrayInputStream;
+import java.io.ByteArrayOutputStream;
+import java.io.DataInputStream;
+import java.io.DataOutputStream;
+import java.io.IOException;
+import java.io.Serializable;
+import java.io.UnsupportedEncodingException;
+import java.nio.ByteBuffer;
+import java.util.HashMap;
+import java.util.Map;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+
+/**
+ * Custom implementation of {@link com.linkedin.pinot.common.utils.DataTableSerDe} interface.
+ * Overrides ser/de methods from {@link DataTableJavaSerDe}
+ */
+public class DataTableCustomSerDe extends DataTableJavaSerDe {
+  private static final Logger LOGGER = LoggerFactory.getLogger(DataTableCustomSerDe.class);
+
+  @Override
+  public byte[] serialize(Object object) {
+    return serializeObject(object);
+  }
+
+  @Override
+  public <T extends Serializable> T deserialize(byte[] bytes, DataType dataType) {
+    return deserializeObject(bytes, dataType);
+  }
+
+  @SuppressWarnings("unchecked")
+  public static <T extends Serializable> T deserializeObject(byte[] bytes, DataType dataType) {
+    ByteBuffer byteBuffer;
+
+    switch (dataType) {
+      case String:
+        try {
+          return (T) new String(bytes, "UTF-8");
+        } catch (UnsupportedEncodingException e) {
+          LOGGER.error("Exception caught while de-serializing String", e);
+          throw new RuntimeException("String de-serialization error");
+        }
+
+      case MutableLong:
+        byteBuffer = ByteBuffer.wrap(bytes);
+        return (T) new MutableLongValue(byteBuffer.getLong());
+
+      case Double:
+        byteBuffer = ByteBuffer.wrap(bytes);
+        return (T) new Double(byteBuffer.getDouble());
+
+      case DoubleArrayList:
+        return (T) deserializeDoubleArray(bytes);
+
+      case AvgPair:
+        return (T) AvgAggregationFunction.AvgPair.fromBytes(bytes);
+
+      case MinMaxRangePair:
+        return (T) MinMaxRangeAggregationFunction.MinMaxRangePair.fromBytes(bytes);
+
+      case HyperLogLog:
+        try {
+          return (T) HyperLogLog.Builder.build(bytes);
+        } catch (IOException e) {
+          LOGGER.error("Exception caught while de-serializing HyperLogLog", e);
+          throw new RuntimeException("HyperLogLog de-serializing error");
+        }
+
+      case QuantileDigest:
+        return (T) deserializeQuantileDigest(bytes);
+
+      case HashMap:
+        return (T) deserializeHashMap(bytes);
+
+      case IntOpenHashSet:
+        return (T) deserializeIntOpenHashSet(bytes);
+
+      case Object:
+        return (T) deserializeJavaObject(bytes);
```

Comment:
Agreed, but this seems to require plumbing through 'metrics' into DataTable from up the stack. Will address that in a separate PR.